include/arch/arm/armv/armv8-a/64/armv/vcpu.h

```diff
@@ -26,7 +26,14 @@
 /* Allow native tasks to run at EL0, but restrict access */
 #define HCR_NATIVE ( HCR_COMMON | HCR_TGE | HCR_TVM | HCR_TTLB | HCR_DC \
                    | HCR_TAC | HCR_SWIO |  HCR_TSC | HCR_IMO | HCR_FMO | HCR_AMO)
+
+#ifdef CONFIG_ALLOW_SMC_CALLS
+/* This should only be a temporary fix. Guests should be able to call into EL3. These
+ * SMC calls should be virtualized in the VMM */
+#define HCR_VCPU   ( HCR_COMMON )
```

Comment:
I'm curious - what SMC calls is your guest using on the zynq? Does this commit help at all: https://github.com/SEL4PROJ/seL4_projects_libs/pull/8/commits/3d88add95ddb8f3432ace455bad57edff52518fd

Reply:
So I cherry-picked your commit, and while that does prevent any VMM issues from triggering, the guest does not work.

The zynqmp needs SMC calls for configuring clocks and other hardware resources. 

Here is the panic that I get from the guest:
```[    0.051622] Kernel panic - not syncing: zynqmp_plat_init power management API version error. Expected: v0.3 - Found: v65535.65535
[    0.051622]
[    0.064421] CPU: 0 PID: 1 Comm: swapper/0 Not tainted 4.9.0-xilinx-v2017.3 #2
[    0.071453] Hardware name: ZynqMP ZCU102 Rev1.0 (DT)
[    0.076344] Call trace:
[    0.078755] [<ffffff8008088138>] dump_backtrace+0x0/0x198
[    0.084069] [<ffffff80080882e4>] show_stack+0x14/0x20
[    0.089046] [<ffffff80083de514>] dump_stack+0x94/0xb8
[    0.094023] [<ffffff800812e9a8>] panic+0x114/0x25c
[    0.098741] [<ffffff8008c53aa8>] zynqmp_plat_init+0x140/0x168
[    0.104402] [<ffffff80080830b8>] do_one_initcall+0x38/0x128
[    0.109894] [<ffffff8008c30bd8>] kernel_init_freeable+0x84/0x1e0
[    0.115814] [<ffffff8008946548>] kernel_init+0x10/0x100
[    0.120961] [<ffffff8008082e80>] ret_from_fork+0x10/0x50
[    0.126198] ---[ end Kernel panic - not syncing: zynqmp_plat_init power management API version error. Expected: v0.3 - Found: v65535.65535
[    0.126198]```